.github/workflows/lint.yml

```diff
@@ -9,12 +9,20 @@ jobs:
   golangci-lint:
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
-          go-version: '1.20'
-      - uses: actions/checkout@v4
+          go-version-file: 'go.mod'
+      # This step sets up the variable steps.golangci-lint-version.outputs.v
+      # to contain the version of golangci-lint (e.g. v1.54.2).
+      # The version is taken from go.mod.
+      - name: Golangci-lint version
+        id: golangci-lint-version
+        run: |
+          GOLANGCI_LINT_VERSION=$( go list -m -f '{{.Version}}' github.com/golangci/golangci-lint )
+          echo "v=$GOLANGCI_LINT_VERSION" >> "$GITHUB_OUTPUT"
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.7.0
+        uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.54.2
-          args: --timeout=5m
```

Comment:
If we don't increase the timeout the lint action might fail randomly, we should add this back.

Sidenote - some other repos include the timeout in [`.golangci.yml`](https://github.com/ConduitIO/conduit/blob/7f64ff1e351f40ab6b3074b0253cd511992d989c/.golangci.yml#L2C7-L2C7), I don't have a preference which approach we use, as long as we increase the timeout, as the runners are sometimes relatively slow and can exceed the default 1m timeout.